tools/normalize_protocol_split_rows.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * This script is the first step to normalizing the  table for projects which store
+ * comma-separated values. This script splits up cells containing commas by creating new rows for
+ * each value and maintaining unique combinations of values.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Liza Levitis <llevitis.mcin@gmail.com>
+ * @license  Loris license
+ * @link     https://www.github.com/aces/Loris
+ */
+
+require_once 'generic_includes.php';
+require_once 'Database.class.inc';
+require_once 'Utility.class.inc';
+
+$DB = \Database::singleton();
+
+/*
+ * Converts an array of arrays of values for different fields into
+ * an array of unique combinations of the different values.
+ */
+function cartesian($input) {
+    // filter out empty values
+    $result = array(array());
+    foreach ($input as $key => $values) {
+        $append = array();
+        foreach($result as $product) {
+            foreach($values as $item) {
+                $product[$key] = $item;
+                // remove $key with empty value so that
+                // an empty string isn't inserted into the database instead of a default NULL value
+                if ($product[$key] == "") {
+                    unset($product[$key]);
+                }
+                $append[] = $product;
+            }
+        }
+        $result = $append;
+    }
+    return $result;
+}
+
+// Get the list of unique IDs from the mri_protocol table
+$mp_idx = $DB->pselect("SELECT ID FROM mri_protocol", array());
```

Comment:
Since you are getting the whole table anyway on line 56, I would refactor this like so:

```php
$rows = $DB->pselect("SELECT ID FROM mri_protocol", array());

foreach ($rows as $row) {
    foreach($row as $key => $value) {
    .
    .
    .
```
and replace `$id` on line 67 with `$value['ID']`. What do you think?